src/calendar/view/CalendarAgendaItemView.ts

```diff
@@ -14,31 +16,32 @@ interface CalendarAgendaItemViewAttrs {
 	color: string
 	click: (domEvent: MouseEvent) => unknown
 	selected?: boolean
+	displayTimeIndicator?: boolean
 }
 
 export class CalendarAgendaItemView implements Component<CalendarAgendaItemViewAttrs> {
 	view({ attrs }: Vnode<CalendarAgendaItemViewAttrs>): Children {
-		return m(
-			".flex.items-center.gap-vpad.click.plr.border-radius.pt-s.pb-s",
-			{
-				class: styles.isDesktopLayout() ? "" : "state-bg",
-				onclick: attrs.click,
-				style: {
-					transition: `background ${DefaultAnimationTime}ms`,
-					background: styles.isDesktopLayout() ? (attrs.selected ? stateBgHover : theme.list_bg) : undefined,
-				},
-			},
-			[
-				m("", {
+		return [
+			m(
+				".flex.items-center.gap-vpad.click.plr.border-radius.pt-s.pb-s.rel",
+				{
+					class: styles.isDesktopLayout() ? "" : "state-bg",
+					onclick: attrs.click,
 					style: {
-						minWidth: "16px",
-						minHeight: "16px",
-						borderRadius: "50%",
-						backgroundColor: `#${attrs.color}`,
+						transition: `background ${DefaultAnimationTime}ms`,
+						background: styles.isDesktopLayout() ? (attrs.selected ? stateBgHover : theme.list_bg) : undefined,
 					},
-				}),
-				m(".flex.col", [m(".b", attrs.event.summary), m("", formatEventTimes(attrs.day, attrs.event, attrs.zone, isAllDayEvent(attrs.event)))]),
-			],
-		)
+				},
+				[
+					attrs.displayTimeIndicator ? m(CalendarTimeIndicator, { offset: -size.vpad_xs }) : null,
```

Comment:
I don't think that indicator should be part of the item view tbh. We should render it once and offset it from top of the list absolutely. This way it will even work if we use the real `List` in agenda.

Reply:
What do you mean by the real `List`? Are you suggesting to calculate the Y offset of the element for the next event in the agenda relative to the top of the list?

Reply:
There's a component called `List` that's usually used for big lists, where we need virtual elements.

I am suggesting we render the indicator on top of the whole list, relative to the top of the list and not inside a `AgendaItemView`.